packages/api-request-builder/test/query-location.spec.js

```diff
@@ -0,0 +1,59 @@
+import * as queryLocation from '../src/query-location'
+import { getDefaultQueryParams } from '../src/default-params'
+
+describe('queryLocation', () => {
+  let service
+
+  beforeEach(() => {
+    service = { params: getDefaultQueryParams(), ...queryLocation }
+  })
+
+  it('should set the country param', () => {
+    service.byCountry('DE')
+    expect(service.params.location.country).toEqual('DE')
+  })
+
+  it('should set the currency param', () => {
+    service.byCountry('DE')
+    service.byCurrency('EUR')
+    expect(service.params.location.currency).toEqual('EUR')
+  })
+
+  it('should set the state param', () => {
+    service.byCountry('DE')
+    service.byState('Germany')
+    expect(service.params.location.state).toEqual('Germany')
+  })
+
+  it('should throw if country is missing', () => {
+    expect(() => service.byCountry()).toThrowError(
+      /Required argument for `byCountry` is missing/
+    )
+  })
+
+  it('should throw if currency is missing', () => {
+    expect(() => service.byCurrency()).toThrowError(
+      /Required argument for `byCurrency` is missing/
+    )
+  })
+
+  it('should throw if state is missing', () => {
+    expect(() => service.byState()).toThrowError(
+      /Required argument for `byState` is missing/
+    )
+  })
+
+  it('should throw while using byCurrency if country was not set', () => {
+    expect(() => service.byCurrency('EUR')).toThrowError(
+      // eslint-disable-next-line max-len
+      /A `country` for this resource has not been set. You must set the country in order to use the `byCurrency` method./
```

Comment:
You can truncate this string to just the first sentence to make lint happy 😉

Reply:
yup yup yup https://github.com/commercetools/nodejs/pull/381/commits/d86d95a6e56544a34f4251ebc8b019391699cc57#diff-abcb522cc38076b7c7e7042c98e38978R49